packages/mobile/components/ActivityTile.svelte

```diff
@@ -1,56 +1,63 @@
 <script lang="typescript">
-    import {
-        Activity,
-        ActivityAsyncStatus,
-        ActivityType,
-        getAssetFromPersistedAssets,
-        InclusionState,
-        IPersistedAsset,
-        selectedAccountAssets,
-    } from '@core/wallet'
+    import { isStrongholdUnlocked } from '@core/profile-manager'
+    import { Activity, ActivityAsyncStatus, ActivityType, claimActivity, InclusionState } from '@core/wallet'
     import {
         AliasActivityTileContent,
-        AsyncActivityTileFooter,
         ClickableTile,
         FoundryActivityTileContent,
         NftActivityTileContent,
         TimelockActivityTileFooter,
         TransactionActivityTileContent,
     } from 'shared/components'
+    import { AsyncActivityTileFooter } from '../components'
+    import { activityRouter, ActivityRoute } from '../lib/routers'
+    import { selectedActivity } from '../lib/wallet'
 
     export let activity: Activity
     export let onClick: () => unknown = () => {}
 
-    let asset: IPersistedAsset
-    $: $selectedAccountAssets, (asset = getAssetFromPersistedAssets(activity.data.assetId))
+    function handleOnClick(): void {
+        $selectedActivity = activity
+        $activityRouter?.goTo(ActivityRoute.Details)
+        onClick()
+    }
```

Comment:
To make the component reusable in the future possibily with desktop, I wouldnt handle this logic here but let the view be the one to decide, so ideally we should add a `onActivityClick` to `packages/mobile/components/ActivityList.svelte` and handle the logic directlty in `ActivityList`, I think it might be better this way